src/components/forms/JurisdictionMetadata/index.tsx

```diff
@@ -0,0 +1,170 @@
+import Papaparse from 'papaparse';
+import React, { useState } from 'react';
+import { Redirect } from 'react-router';
+import { toast } from 'react-toastify';
+import { Button, Label } from 'reactstrap';
+import { FormGroup } from 'reactstrap';
+import * as Yup from 'yup';
+import {
+  CSV_FILE,
+  CSV_ONLY,
+  FILE,
+  FILE_FORMAT,
+  FILE_UPLOADED_SUCCESSFULLY,
+  INVALID_CSV,
+  REQUIRED,
+  UPLOAD,
+  UPLOAD_FILE,
+  UPLOADING,
+} from '../../../configs/lang';
+import { HOME_URL, OPENSRP_V1_SETTINGS_ENDPOINT } from '../../../constants';
+import { createPayloads, growl, PapaResult, SettingConfiguration } from '../../../helpers/utils';
+import { OpenSRPService } from '../../../services/opensrp';
+
+const SUPPORTED_FORMATS = ['text/csv'];
+
+/** yup validation schema for Jurisdiction Metadata Form input */
+export const JurisdictionSchema = Yup.object().shape({
+  file: Yup.mixed()
+    .required(REQUIRED)
+    .test(FILE_FORMAT, CSV_ONLY, value => value && SUPPORTED_FORMATS.includes(value.type)),
+});
+
+export interface JurisdictionMetadataFormFields {
+  file: File;
+}
+
+export interface JurisdictionMetadataFormProps {
+  disabledFields: string[];
+  OpenSRPService: new (...args: any[]) => any;
+  submitForm: (
+    setSubmitting: (isSubmitting: boolean) => void,
+    setGlobalError: (errorMessage: string) => void,
+    setIfDoneHere: (closeSubmissionCycle: boolean) => void,
+    props: JurisdictionMetadataFormProps,
+    values?: JurisdictionMetadataFormFields
+  ) => void;
+  initialValues: JurisdictionMetadataFormFields;
+  redirectAfterAction: string;
+}
+
+export const defaultInitialValues: JurisdictionMetadataFormFields = {
+  file: new File([], ''),
+};
+
+// read csv and convert to json
+const handleFile = (file: File, complete: (results: PapaResult) => void) => {
+  Papaparse.parse(file, {
+    complete: results => {
+      complete(results);
+    },
+    header: true,
+  });
+};
+
+export const submitForm = (
+  setSubmitting: (isSubmitting: boolean) => void,
+  setGlobalError: (errorMessage: string) => void,
+  setIfDoneHere: (closeSubmissionCycle: boolean) => void,
+  props: JurisdictionMetadataFormProps,
+  values?: JurisdictionMetadataFormFields
+) => {
+  if (values) {
+    const jurisdictionService = new props.OpenSRPService(OPENSRP_V1_SETTINGS_ENDPOINT);
+    handleFile(values.file, results => {
+      const payloads: SettingConfiguration[] = createPayloads(results);
+      if (payloads.length > 0) {
+        const valuesToSend = JSON.stringify({ settingConfigurations: payloads });
+        jurisdictionService
+          .create(valuesToSend)
+          .then(() => {
+            growl(FILE_UPLOADED_SUCCESSFULLY, {
+              onClose: () => setIfDoneHere(true),
+              type: toast.TYPE.SUCCESS,
+            });
+          })
+          .catch((e: Error) => {
+            setGlobalError(e.message);
+          });
+        setSubmitting(false);
+      } else {
+        setGlobalError(INVALID_CSV);
```

Comment:
![what error](https://user-images.githubusercontent.com/372073/85098683-09aa7200-b204-11ea-96fd-055bd067bc37.png)


We need to unset the error once the user selects another CSV file and/or clicks submit.  This is so that the previous error message is not stuck on the screen, with the user not knowing whether it represents a new error or not.